File: go-sdk/pkg/models/operations/getapipagesid.go
package operations

import (
	"github.com/speakeasy-sdks/forem-sdks/go-client-sdk/pkg/models/shared"
)

type GetAPIPagesIDPathParams struct {
	ID int32 `pathParam:"style=simple,explode=false,name=id"`
}

type GetAPIPagesIDRequest struct {
	PathParams GetAPIPagesIDPathParams
}

type GetAPIPagesIDResponse struct {
	ContentType string
	Page        *shared.Page
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/getapipagesid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetApiPagesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}


export class GetApiPagesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetApiPagesIdPathParams;
}


export class GetApiPagesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  page?: shared.Page;

  @SpeakeasyMetadata()
  statusCode: number;
}
